primitives/arithmetic/src/fixed64.rs

```diff
@@ -113,15 +113,33 @@ impl Saturating for Fixed64 {
 	}
 
 	fn saturating_mul(self, rhs: Self) -> Self {
-		Self(self.0.saturating_mul(rhs.0) / DIV)
+		let a = self.0 as i128;
+		let b = rhs.0 as i128;
+		let res = a.saturating_mul(b) / DIV as i128;
```

Comment:
```suggestion
		let res = a * b / DIV as i128;
```

`i64::MAX * i64::MAX` fits into `i128`